### PR TITLE
Pass the geometry dimension to functions for creating properties

### PR DESCRIPTION
**Pull request by wenqing**
As titled,  `ProjectData::_mesh_vec[0]->getDimension()` is passed into the functions for creating properties.

There exists a  property creation function, 
 [CreatePermeabilityOrthotropicPowerLaw.cpp](https://github.com/ufz/ogs/blob/master/MaterialLib/MPL/Properties/CreatePermeabilityOrthotropicPowerLaw.cpp#L64-L82), 
which use the size of the input parameter as the geometry dimension. For properties that such parameter size may not be identical to the the geometry dimension, such consideration is invalid. Besides, the input info should be made as unique as possible. Therefore, the changes in this PR were made.

### CODE REVIEW

File: MaterialLib/MPL/CreateProperty.cpp [hunk 1]
```diff
@@ -27,6 +27,7 @@
 namespace
 {
 std::unique_ptr<MaterialPropertyLib::Property> createProperty(
+    int const /*geometry_dimension*/,
```
Comment (wenqing) [inline MaterialLib/MPL/CreateProperty.cpp:30]: The comment is removed in PR #2989.

File: Tests/MaterialLib/TestMPL.cpp [hunk 2]
```diff
@@ -19,7 +19,8 @@
 #include "MathLib/InterpolationAlgorithms/PiecewiseLinearInterpolation.h"
 #include "ParameterLib/Parameter.h"
 
-std::unique_ptr<MPL::Medium> createTestMaterial(std::string const& xml)
+std::unique_ptr<MPL::Medium> createTestMaterial(
+    std::string const& xml, int const geometry_dimention)
```
Comment (endJunction) [inline Tests/MaterialLib/TestMPL.cpp:23]: I like your humor :)

**Comment by TomFischer** [comment]
```grep -Rl dimention | xargs sed -i 's/dimention/dimension/g'```

**Comment by wenqing** [comment]
Corrected that spelling in TestMPL.h/cpp.

File: MaterialLib/MPL/CreatePhase.h [hunk 3]
```diff
@@ -48,7 +48,8 @@ namespace MaterialPropertyLib
 /// assigned. These default properties average the component properties,
 /// weighted by mole fraction.
 std::vector<std::unique_ptr<Phase>> createPhases(
-    boost::optional<BaseLib::ConfigTree> const& config,
+    int const geometry_dimension,
+        boost::optional<BaseLib::ConfigTree> const& config,
```
Comment (endJunction) [inline MaterialLib/MPL/CreatePhase.h:52]: formatting

**Comment by endJunction** [comment]
Looks OK, but needs clang-formatting. I do understand, that different clang-format version produce slightly different output, though in other conversation you told, that you are using version 10 (which is equal to the installed version on my computer). There are changes (not only in this PR), appearing again and again obviously not being formatted.

There are options to use pre-commit checks. You can also run `git rebase -i -x 'git clang-format HEAD^' ufz/master` *before* making a PR. Your editor might have an option to run clang-format on the changed lines before saving.

For this PR the difference is:
```diff
diff --git a/Applications/ApplicationsLib/ProjectData.cpp b/Applications/ApplicationsLib/ProjectData.cpp
index 57cc58452..01e60a54a 100644
--- a/Applications/ApplicationsLib/ProjectData.cpp
+++ b/Applications/ApplicationsLib/ProjectData.cpp
@@ -486,8 +486,8 @@ void ProjectData::parseMedia(
             _media[id] =
                 (id == material_ids[0])
                     ? MaterialPropertyLib::createMedium(
-                          _mesh_vec[0]->getDimension(),
-                          medium_config, _parameters,
+                          _mesh_vec[0]->getDimension(), medium_config,
+                          _parameters,
                           _local_coordinate_system ? &*_local_coordinate_system
                                                    : nullptr,
                           _curves)
diff --git a/MaterialLib/MPL/CreateComponent.cpp b/MaterialLib/MPL/CreateComponent.cpp
index e220b1122..12979f29e 100644
--- a/MaterialLib/MPL/CreateComponent.cpp
+++ b/MaterialLib/MPL/CreateComponent.cpp
@@ -91,9 +91,9 @@ std::vector<std::unique_ptr<Component>> createComponents(
         //! \ogs_file_param{prj__media__medium__phases__phase__components__component}
         config->getConfigSubtreeList("component"))
     {
-        auto component = createComponent(geometry_dimension,
-                                         component_config, parameters,
-                                         local_coordinate_system, curves);
+        auto component =
+            createComponent(geometry_dimension, component_config, parameters,
+                            local_coordinate_system, curves);
 
         if (std::find_if(components.begin(),
                          components.end(),
diff --git a/MaterialLib/MPL/CreatePhase.h b/MaterialLib/MPL/CreatePhase.h
index 5ba7d1e13..e956d4b5d 100644
--- a/MaterialLib/MPL/CreatePhase.h
+++ b/MaterialLib/MPL/CreatePhase.h
@@ -49,7 +49,7 @@ namespace MaterialPropertyLib
 /// weighted by mole fraction.
 std::vector<std::unique_ptr<Phase>> createPhases(
     int const geometry_dimension,
-        boost::optional<BaseLib::ConfigTree> const& config,
+    boost::optional<BaseLib::ConfigTree> const& config,
     std::vector<std::unique_ptr<ParameterLib::ParameterBase>> const& parameters,
     ParameterLib::CoordinateSystem const* const local_coordinate_system,
     std::map<std::string,
diff --git a/Tests/MaterialLib/TestMPL.cpp b/Tests/MaterialLib/TestMPL.cpp
index 585885fe0..07bf268da 100644
--- a/Tests/MaterialLib/TestMPL.cpp
+++ b/Tests/MaterialLib/TestMPL.cpp
@@ -22,8 +22,8 @@
 
 namespace Tests
 {
-std::unique_ptr<MPL::Medium> createTestMaterial(
-    std::string const& xml, int const geometry_dimension)
+std::unique_ptr<MPL::Medium> createTestMaterial(std::string const& xml,
+                                                int const geometry_dimension)
 {
     auto const ptree = Tests::readXml(xml.c_str());
     BaseLib::ConfigTree conf(ptree, "", BaseLib::ConfigTree::onerror,
```

**Comment by wenqing** [comment]
> Looks OK, but needs clang-formatting. I do understand, that different clang-format version produce slightly different output, though in other conversation you told, that you are using version 10 (which is equal to the installed version on my computer). There are changes (not only in this PR), appearing again and again obviously not being formatted.

Thanks. It is weird. I use clang-format  in my editor (which seems not from system for such difference), and use git-clange-format as well. From now on, use  git-clange-format  only.

**Comment by endJunction** [comment]
This new clang-format setup works perfectly well. Thanks for taking time to adjust it.
:fast_forward: when green.

**Pull request merged**